src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -358,6 +358,101 @@ AccountRootsNotDeleted::finalize(
 
 //------------------------------------------------------------------------------
 
+void
+AccountRootsDeletedClean::visitEntry(
+    bool isDelete,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const&)
+{
+    if (isDelete && before && before->getType() == ltACCOUNT_ROOT)
+        accountsDeleted_.emplace_back(before);
+}
+
+bool
+AccountRootsDeletedClean::finalize(
+    STTx const& tx,
+    TER const result,
+    XRPAmount const,
+    ReadView const& view,
+    beast::Journal const& j)
+{
+    // This list should include all of the keylet functions that take a single
+    // AccountID parameter, except nftpage_max, because it is exercised below
+    // using the NFT page lookup logic. (nftpage_min is included because it
+    // should _never_ be present.)
+    using fType = std::function<Keylet(AccountID const& id)>;
+    static std::array<fType, 4> const keyletfuncs{
+        &keylet::account,
+        &keylet::ownerDir,
+        &keylet::signers,
+        &keylet::nftpage_min,
+        //&keylet::nftpage_max
```

Comment:
Should this line be removed?

Reply:
I centralized the list of functions into `Indexes.h`, included all the functions, and rewrote the invariant slightly to not duplicate effort.